app/models/case/sar/standard.rb

```diff
@@ -146,11 +151,20 @@ def reset_deadline!
     )
   end
 
-  # SARs extensions are based on calendar days not working days
+  # The deadlines are all calculated based on the date receiving the case
   def max_allowed_deadline_date
-    initial_deadline + Settings.sar_extension_limit.to_i.days
+    time_limit = self.correspondence_type.extension_time_limit || Settings.sar_extension_default_limit
```

Comment:
I wonder if you could make this line into another private method called `max_time_limit` ?

Reply:
Could do!